zerver/lib/actions.py

```diff
@@ -4622,7 +4622,8 @@ def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
     move_messages_to_archive(message_ids)
     for event, users_to_notify in events_and_users_to_notify:
         # TODO: Figure out some kind of bulk event that we could send just one of?
-        send_event(realm, event, users_to_notify)
+        if users_to_notify:
+            send_event(realm, event, users_to_notify)
```

Comment:
Hrm, I think technically it could be wrong to not send an event even if there are no subscribers, in that we do have a (very rarely used) feature allowing an API client to request "all new messages for stream X" without subscribing to the stream, if it's public.  

So perhaps it'd be better to just make it work to send an empty list of users for that reason?

Reply:
Ah, makes sense, I dropped the commit